src/advanced_agent/database/connection.py

```
@@ -0,0 +1,295 @@
+        if not self.SessionLocal:
+            raise RuntimeError("データベースが初期化されていません")
+        return self.SessionLocal()
+    
+    def execute_raw_sql(self, sql: str, params: Optional[Dict[str, Any]] = None) -> Any:
+        """生SQL実行"""
+        try:
+            with self.get_session() as session:
+                result = session.execute(text(sql), params or {})
+                return result.fetchall()
+        except Exception as e:
+            logger.error(f"生SQL実行エラー: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)